lib/messages/commands/tx.js

```diff
@@ -2,28 +2,30 @@
 
 var Message = require('../message');
 var inherits = require('util').inherits;
+var bitcore = require('bitcore');
+var $ = bitcore.util.preconditions;
+var _ = bitcore.deps._;
 
 /**
- * @param {Object|Transaction=} options - If is an instance of Transaction will use as options.transaction
- * @param {Transaction=} options.transaction - An instance of a Transaction
+ * @param {Transaction=} arg - An instance of Transaction
+ * @param {Object} options
  * @param {Number} options.magicNumber
  * @extends Message
  * @constructor
  */
-function TransactionMessage(options) {
-  Message.call(this, options);
+function TransactionMessage(arg, options) {
+  Message.call(this, arg, options);
   this.command = 'tx';
   this.magicNumber = options.magicNumber;
   this.Transaction = options.Transaction;
-
-  var transaction;
-  if(options instanceof this.Transaction) {
-    transaction = options;
-  } else {
-    transaction = options.transaction;
+  $.checkArgument(
+    _.isUndefined(arg) || arg instanceof this.Transaction,
+    'An instance of Transaction or undefined is expected'
```

Comment:
Mostly because of this test: https://github.com/braydonf/bitcore-p2p/blob/bug/transactions-message/test/messages/index.js#L58 I'll take a look and see if it's necessary.